### PR TITLE
Megatron-lm checkpoints need to have version set.

### PR DESCRIPTION
**Pull request by ericharper**
Checking for version in restore and then setting it. 

Need to save checkpoint version when saving NeMo Megatron-lm models.

### CODE REVIEW

File: nemo/collections/nlp/modules/common/megatron/megatron_bert.py [hunk 1]
```diff
@@ -151,6 +157,11 @@ def restore_weights(self, restore_path: str):
                 mp_restore_path = f'{restore_path}/mp_rank_{model_parallel_rank:02d}/model_optim_rng.pt'
                 logging.info(f'Restoring model parallel checkpoint from: {mp_restore_path}')
                 state_dict = torch.load(mp_restore_path, map_location='cpu')
+                if 'checkpoint_version' in state_dict:
+                    if state_dict['checkpoint_version'] is not None:
+                        set_checkpoint_version(state_dict['checkpoint_version'])
+                else:
+                    set_checkpoint_version(0)
```
Comment (ekmb) [inline nemo/collections/nlp/modules/common/megatron/megatron_bert.py:164]: Is the `checkpoint_version` added to the state_dict during this call?

File: nemo/collections/nlp/modules/common/megatron/megatron_bert.py [hunk 2]
```diff
@@ -151,6 +157,11 @@ def restore_weights(self, restore_path: str):
                 mp_restore_path = f'{restore_path}/mp_rank_{model_parallel_rank:02d}/model_optim_rng.pt'
                 logging.info(f'Restoring model parallel checkpoint from: {mp_restore_path}')
                 state_dict = torch.load(mp_restore_path, map_location='cpu')
+                if 'checkpoint_version' in state_dict:
+                    if state_dict['checkpoint_version'] is not None:
+                        set_checkpoint_version(state_dict['checkpoint_version'])
```
Comment (ekmb) [inline nemo/collections/nlp/modules/common/megatron/megatron_bert.py:162]: Should we throw an error/warning if this check fails?

**Review by ekmb — APPROVED** [review]

File: nemo/collections/nlp/modules/common/megatron/megatron_bert.py (hunk 1 above)
Comment (ericharper) [reply nemo/collections/nlp/modules/common/megatron/megatron_bert.py:164]: No, it's not, I'm still trying to figure out how to persist it when we save on the NeMo side.

File: nemo/collections/nlp/modules/common/megatron/megatron_bert.py (hunk 2 above)
Comment (ericharper) [reply nemo/collections/nlp/modules/common/megatron/megatron_bert.py:162]: Yes, i'll add a warning message noting that we're defaulting to checkpoint_version=0.

**Pull request merged**